### ISSUE TITLE
ENH: provide numpy.random.shuffle(a, b, ...), 2 arrays in unison

### ISSUE THREAD

**Issue by mw66** [numpy/numpy#8204]
http://stackoverflow.com/questions/4601373/better-way-to-shuffle-two-numpy-arrays-in-unison

numpy.random.shuffle only shuffle 1 array in-place.

It will be better to provide another function to shuffle 2 / multiple arrays in-place, and in unison.

I like the proposed:

def shuffle_in_unison_scary(a, b):
    rng_state = numpy.random.get_state()
    numpy.random.shuffle(a)
    numpy.random.set_state(rng_state)
    numpy.random.shuffle(b)

And want numpy guys to confirm this implementation, and add to numpy library itself.

**Comment by wrwrwr** [comment]
Wouldn't it be natural to let [`random.shuffle`](https://docs.scipy.org/doc/numpy/reference/generated/numpy.random.shuffle.html) take more arguments as in [`sklearn.utils.shuffle`](http://scikit-learn.org/stable/modules/generated/sklearn.utils.shuffle.html)? Wouldn't the `RandomState` need to be made thread-local?

**Comment by mingwugmail** [comment]
Take more or same arguments as in sklearn.utils.shuffle is fine.

But sklearn.utils.shuffle use the double amount of memory in my test. I'd like the in-place implementation without using 2x times memory of the input arrays.

**Comment by andanis** [comment]
http://rasbt.github.io/mlxtend/user_guide/preprocessing/shuffle_arrays_unison/

**Comment by WarrenWeckesser** [comment]
There has been no activity here from core devs for more than five years, so I think it is pretty clear that there is no interest in adding such functionality to NumPy.  I'm closing the issue, but if it turns out that there *is* interest, it can be reopened.
